File: backend/src/main/java/com/example/Flicktionary/global/service/EmailService.java
package com.example.Flicktionary.global.service;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class EmailService {

    private final JavaMailSender javaMailSender;

    @Value("${custom.mail.sender}")
    private final String emailSenderAddress;

    public void sendSimpleEmail(String to, String subject, String content) {
        SimpleMailMessage message = new SimpleMailMessage();
        message.setFrom(emailSenderAddress);
        message.setTo(to);
        message.setSubject(subject);
        message.setText(content);
        javaMailSender.send(message);
    }
}

File: backend/src/main/java/com/example/Flicktionary/global/service/EmailService.kt
package com.example.Flicktionary.global.service

import org.springframework.beans.factory.annotation.Value
import org.springframework.mail.SimpleMailMessage
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.stereotype.Service

@Service
class EmailService(
    private val javaMailSender: JavaMailSender
) {
    @Value("\${custom.mail.sender}")
    private lateinit var emailSenderAddress: String

    fun sendSimpleEmail(to: String, subject: String, content: String) {
        val message = SimpleMailMessage()
        message.from = emailSenderAddress
        message.setTo(to)
        message.subject = subject
        message.text = content
        javaMailSender.send(message)
    }
}